task/host-sp-comms/src/bsp/gimletlet.rs

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! SP inventory types and implementation
+//!
+//! This reduces clutter in the main `ServerImpl` implementation
+use super::ServerImpl;
+use host_sp_messages::{InventoryData, InventoryDataResult};
+
+impl ServerImpl {
+    /// Number of devices in our inventory
+    pub(crate) const INVENTORY_COUNT: u32 = 1;
+
+    pub(crate) fn perform_inventory_lookup(
+        &mut self,
+        sequence: u64,
+        index: u32,
+    ) -> Result<(), InventoryDataResult> {
+        #[forbid(unreachable_patterns)]
+        match index {
+            0 => {
+                // U12: the service processor itself
+                // The UID is readable by stm32xx_sys
+                let sys =
+                    drv_stm32xx_sys_api::Sys::from(crate::SYS.get_task_id());
+                let uid = sys.read_uid();
+                let data = InventoryData::Stm32H7 {
+                    uid,
+                    dbgmcu_rev_id: 0, // TODO
+                    dbgmcu_dev_id: 0, // TODO
```

Comment:
Just double-checking - did you want to address these `TODO`s before merging?

Reply:
Yes, done in 84a157c8c6ca191fdd8181aee8a00bc6fca01969